### ISSUE TITLE
Add MacOS Support

### ISSUE THREAD

**Issue by dblock** [opensearch-project/sql#250]
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/96, add MacOS (Darwin) support.

- [ ] Passing CI on MacOS
- [ ] Documentation

**Comment by derek-ho** [comment]
Waiting for review on: https://github.com/opensearch-project/sql/pull/907. then will close this ticket

**Comment by derek-ho** [comment]
Merged into 2.x, waiting on backport PR to go in before I close this issue

**Comment by derek-ho** [comment]
Merged in PR and backport PR so closing this issue:
https://github.com/opensearch-project/sql/pull/967
https://github.com/opensearch-project/sql/pull/907
https://github.com/opensearch-project/sql/pull/962
